### PR TITLE
Add custom volume mappings

### PR DESCRIPTION
**Pull request by goneflyin**
Here are the changes to the mix-edip extension used by the EDIP docker image changes submitted in https://github.com/asaaki/docker-images/pull/1. 

As I've updated the documentation on how to use the new option, I _think_ the code can stand by itself with no further explanation. That said, if you have questions or changes, please let me know! Thanks!

### CODE REVIEW

File: lib/edip/options.ex [hunk 1]
```diff
@@ -4,33 +4,67 @@ defmodule Edip.Options.Package do
...
+  def parse_args(args) do
+    {parsed_opts, _, _} = OptionParser.parse(args, aliases: @option_aliases, strict: @option_stricts)
+    |> IO.inspect
+    |> consolidate_mappings
+
+    parsed_opts
   end
 
-  defp writer(valid_options) do
+  def writer(valid_options) do
```
Comment (asaaki) [inline lib/edip/options.ex]: Is there a reason why this function is public now? Do you had any issues?

File: lib/edip/options.ex [hunk 2]
```diff
@@ -4,33 +4,67 @@ defmodule Edip.Options.Package do
...
+    |> Keyword.get_values(:mapping)
+    |> Enum.map(&create_valid_mapping/1)
+
+    parsed = parsed |> Keyword.delete(:mapping) |> Keyword.merge(mappings: mappings)
+
+    {parsed, argv, errors}
+  end
+
+  def create_valid_mapping(mapping_opt) do
+    case mapping_opt |> String.split(":") do
```
Comment (asaaki) [inline lib/edip/options.ex:65]: There can be actually 2 `:` in a mapping:

```
from:to:access_options

access_options = rw | ro | ...
```

We should take this into account and respect such options, too.
Would you update and extend this?

File: lib/edip/options.ex [hunk 3]
```diff
@@ -4,33 +4,67 @@ defmodule Edip.Options.Package do
...
     package = struct(Package, valid_options)
-    struct(__MODULE__, Dict.merge(valid_options, [writer: writer, package: package]))
+    mappings = valid_options[:mappings]
+    struct(__MODULE__, Dict.merge(valid_options, [writer: writer, package: package, mappings: mappings]))
   end
 
-  defp parse_args(args) do
-    {valid_opts, _, _} = OptionParser.parse(args, aliases: @option_aliases, strict: @option_stricts)
-    valid_opts
+  def parse_args(args) do
```
Comment (asaaki) [inline lib/edip/options.ex]: Is there a reason why this function is public now? Do you had any issues?

File: lib/edip/runner.ex [hunk 4]
```diff
@@ -4,8 +4,10 @@ defmodule Edip.Runner do
   alias Edip.ImageConfig
   alias Edip.Settings
 
-  @edip_version          "0.3.0"
```
Comment (asaaki) [inline lib/edip/runner.ex:7]: You can reverse this change and update to version `0.4.2` now.

File: lib/edip/options.ex (hunk 1 above)
Comment (goneflyin) [inline lib/edip/options.ex]: d'oh -- nope! I think I was checking something and forgot to revert it... You know, I had to do some temporary changes to move forward on some local dev -- and I didn't think to do it on a separate branch. Sorry if there's noise! I'll review and get this sort of thing removed entirely.

File: lib/edip/options.ex (hunk 2 above)
Comment (goneflyin) [inline lib/edip/options.ex:65]: Ahhh....great catch! Yes, I'll do so.

File: lib/edip/options.ex (hunk 3 above)
Comment (goneflyin) [inline lib/edip/options.ex]: Same (stupid) reason as above. Will fix!

File: lib/edip/runner.ex (hunk 4 above)
Comment (goneflyin) [inline lib/edip/runner.ex:7]: Excellent! I'll take care of that shortly.

**Comment by goneflyin** [comment]
All of the points you raised in your comments should be addressed now. As an extra bonus, I've added some tests for the `Edip.Options` and `Edip.Settings` modules where I needed to make changes. 

Let me know if there's still anything that needs addressing!

File: test/edip/settings_test.exs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    assert String.contains?(pkg_options, "RELEASE_PREFIX=prefixer")
+  end
+
+  test "basic mapping is formatted for docker run command" do
+    mapping_options =
+      [%Options.Mapping{from: "/from/vol", to: "/to/vol"}]
+    |> Edip.Settings.from_mapping_options
+
+    assert mapping_options == "-v \"/from/vol:/to/vol\""
```
Comment (asaaki) [inline test/edip/settings_test.exs]: [Hint] If you want to get rid of the backslash escaping for double quotes you can use the `~s()` instead:

``` elixir
~s(-v "/from/vol:/to/vol")
```

**Comment by asaaki** [comment]
Looks good to me.

Could you please squash all the commits into a single one?
(because there are some temp, test and revert commits, which doesn't read so nicely in the history)

Then we're good to go and I can craft a new release soon.

**Comment by goneflyin** [comment]
Updated tests and squashed commits. Should be good to go!

**Comment by asaaki** [comment]
Thank you for your help! :bow:

**Pull request merged**